config.py

```diff
@@ -1421,8 +1421,106 @@ class ArbitraryContrast(TypedDict):
 epochs before passing them to time-frequency analysis. Set this to `True` to
 highlight induced activity.
+
+Note: Note
+    This also applies to CSP analysis.
+"""
+
+###############################################################################
+# TIME-FREQUENCY CSP
+# ------------------
+
+decoding_csp: bool = False
+"""
+Whether to run decoding via Common Spatial Patterns (CSP) analysis on the
+time-frequency representation of the data.
+"""
+
+decoding_csp_times: Optional[ArrayLike] = np.linspace(
+    max(0, epochs_tmin),
+    epochs_tmax,
+    num=6
+)
+"""
+The edges of the time bins to use for CSP decoding.
+Must contain at least two elements. By default, 4 equally-spaced bins are
+created across the non-negative time range of the epochs.
+
+All specified time points must be contained in the epochs interval.
+
+If `None`, do not perform time-frequency analysis, and only run CSP on
+frequency data.
+
+???+ example "Example"
+    Create 3 equidistant time bins (0–0.2, 0.2–0.4, 0.4–0.6 sec):
+    ```python
+    decoding_csp_times = np.linspace(start=0, stop=0.6, num=4)
+    ```
+    Create 2 time bins of different durations (0–0.4, 0.4–0.6 sec):
+    ```python
+    decoding_csp_times = [0, 0.4, 0.6]
+    ```
+"""
+
+decoding_csp_freqs: Dict[str, ArrayLike] = {
+    'custom': [
+        time_frequency_freq_min,
+        (time_frequency_freq_max - time_frequency_freq_min) / 2 + time_frequency_freq_min,  # noqa: E501
+        time_frequency_freq_max
+    ]
+}
+"""
+The edges of the frequency bins to use for CSP decoding.
+
+This parameter must be a dictionary with they keys specifying the "name" to use
+for the frequency range, must be a list-like object containing at least two
+scalar values, specifying the edges of the respective frequency bin(s).
+
+Defaults to two frequency bins, one from
+[`time_frequency_freq_min`][config.time_frequency_freq_min] to the mindpoint
+between this value and
+[`time_frequency_freq_max`][config.time_frequency_freq_max]; and the other from
+that midpoint to `time_frequency_freq_max`.
+
+???+ example "Example"
+    Create two frquency bins, one for 4–8 Hz, and another for 8–14 Hz:
+    ```python
+    decoding_csp_freqs = {
+        'custom_range': [4, 8, 14]
+    }
+    ```
+
+    Create 5 equidistant frequency bins from 4 to 14 Hz:
+    ```python
+    decoding_csp_freqs = {
+        'custom_range': np.linspace(
+            start=4,
+            stop=14,
+            num=5+1  # We need one more to account for the endpoint!
+        )
+    }
+    ```
+
+    Create two frquency bins with a "gap" between them, one for 4–8 Hz,
+    and another for 10–14 Hz:
+    ```python
+    decoding_csp_freqs = {
+        'lower_range': [4, 8],
+        'upper_range': [10, 14]
+    ]
+    ```
+
+    Create partially overlapping frequency bins, one for 4–10 Hz, and another
+    for 8–14 Hz:
+    ```python
+    decoding_csp_freqs = {
+        'lower_range': [4, 10],
+        'upper_range': [8, 14]
```

Comment:
how does this affect 2d plots?

Reply:
For each freq range, we basically perform an independent analysis, and also generate independent figures. See here where I had an "alpha" and a "test" freq range.

<img width="1008" alt="Screen Shot 2022-07-20 at 14 26 22" src="https://user-images.githubusercontent.com/2046265/179981606-6593fde0-6eee-489d-9d74-06049be47894.png">

Reply:
For each freq range, we basically perform an independent analysis, and also generate independent figures. See here where I had an "alpha" and a "test" freq range.

<img width="1008" alt="Screen Shot 2022-07-20 at 14 26 22" src="https://user-images.githubusercontent.com/2046265/179981606-6593fde0-6eee-489d-9d74-06049be47894.png">